tests/integration/components/async-await-test.js

```diff
@@ -100,6 +100,47 @@ module('Integration | Component | async-await', function(hooks) {
         assert.dom().hasText('resolved value');
       });
 
+      test('it can take a hash of promises as arguments', async function(assert) {
+        this.set('promiseA', Promise.resolve('valueA'));
+        this.set('promiseB', Promise.resolve('valueB'));
+        this.set('promiseC', Promise.resolve('valueC'));
+
+        await render(hbs`
+          {{#async-await (hash a=this.promiseA b=this.promiseB c=this.promiseC) as |h|}}
+            resolved {{h.a}}, {{h.b}}, {{h.c}}
+          {{/async-await}}
+        `);
+
+        assert.dom().hasText('resolved valueA, valueB, valueC');
+      });
+
+      test('it can take a mixed hash as arguments', async function(assert) {
+        this.set('promiseA', Promise.resolve('valueA'));
+        this.set('valueB', 'valueB');
+        this.set('promiseC', Promise.resolve('valueC'));
+
+        await render(hbs`
+          {{#async-await (hash a=this.promiseA b=this.valueB c=this.promiseC) as |h|}}
+            resolved {{h.a}}, {{h.b}}, {{h.c}}
+          {{/async-await}}
+        `);
+
+        assert.dom().hasText('resolved valueA, valueB, valueC');
+      });
+
+      test('it can take an object as the argument', async function(assert) {
+        this.set('value', { foo: 'bar' });
+
+        // Expect a straight pass-through
+        await render(hbs`
+          {{#async-await value as |v|}}
+            resolved {{eq v value}}
+          {{/async-await}}
+        `);
+
+        assert.dom().hasText('resolved true');
+      });
```

Comment:
what about something like:

```
let obj = {
  toString() { return 'fancy object'; }
};

this.set('value', obj);

let captured;

this.owner.register('helper:capture', helper(function([value]) {
  captured = value;
  return value;
});

// Expect a straight pass-through
await render(hbs`
  {{#async-await value as |v|}}
    resolved {{capture v}}
  {{/async-await}}
`);

assert.dom().hasText('resolved object');
assert.strictEqual(captured, obj);
```

The `===`-ness is crucial to this test, and the `eq` helper doesn't make it super clear whether it uses `==` or `===`

Reply:
It is `===` (https://github.com/jmurphyau/ember-truth-helpers#usage) but I like this better.